File: dice/python/sol/apigear_game/dice.py
from apigear_game.api import api

class Dice(api.IDice):
    def __init__(self, random: api.IRandom):
        self.random = random
        self.__value : int = 0

    @property
    def value(self):
        return self.__value

    def toss(self) -> None:
        self.__value = self.random.generate(1, 6)

File: dice/typescript/sol/src/apigear_game/dice.ts
import { IRandom } from "./api";
import * as api from "./api";

export class Dice implements api.IDice {
  value = 0;
  random: IRandom;
  constructor(random: IRandom) {
    this.random = random;
  }
  toss(): void {
    this.value = this.random.generate(1, 6);
  }
}
